perllib/FixMyStreet/Roles/Cobrand/Echo.pm

```diff
@@ -1302,6 +1302,94 @@ sub per_photo_size_limit_for_report_in_bytes {
     return min($max_size_per_image, $max_size_per_image_from_total);
 };
 
+=head2 update_report_for_same_day_amend
+
+This Echo only hook is used to check
+if a Bulky collection that has been changed
+has maintained the same date
+
+=cut
+
+sub update_report_for_same_day_amend {
+        my ($self, $p, $data) = @_;
+
+        my $collection_date = $p->get_extra_field_value('Collection_Date');
+        return $data->{chosen_date} =~ /^$collection_date;/
+};
+
+=head2 waste_munge_bulky_amend
+
+If a Bulky collection has been amended, but the
+date of collection remains the same, we need
+to reset the data in the original report as
+that will continue to reflect the actual
+booking
+
+=cut
+
+sub waste_munge_bulky_amend {
+    my ($self, $p, $data) = @_;
+
+    for my $extra (keys %{$p->get_extra_metadata}) {
+        if ($extra =~ /item_/) {
+            $p->unset_extra_metadata($extra);
+        }
+    }
+    $self->waste_munge_bulky_data($data);
+
+    $self->{c}->stash->{report} = $p;
+    $self->save_item_names_to_report($data);
+
+    my @fields = @{$p->get_extra_fields};
+    for my $field (@fields) {
```

Comment:
Rather than this way round, could this loop over all the fields perhaps be instead:
```
foreach (qw(Bulky_Collection_Bulky_Items Bulky_Collection_Notes
    TEM_-_Bulky_Collection_Description TEM_-_Bulky_Collection_Item Exact_Location)) {
    if ($data->{"extra_$_"}) {
        $p->update_extra_field({ name => $_, value => $data->{"extra_$_"} });
    }
}
```
Does that work okay?

Reply:
Could even put the cobrand specific fields in a cobrand function and then call it for the list here and in your open_311_bulky_update_same_date perhaps